cms/djangoapps/contentstore/tests/test_libraries.py

```diff
@@ -729,6 +741,70 @@ def test_refresh_library_content_permissions(self, library_role, course_role, ex
         lc_block = self._refresh_children(lc_block, status_code_expected=200 if expected_result else 403)
         self.assertEqual(len(lc_block.children), 1 if expected_result else 0)
 
+    def test_studio_user_permissions(self):
+        """
+        Test that user could attach to the problem only libraries that he has access (or which were created by him).
+        This test was created on the basis of bug described in the pull requests on github:
+        https://github.com/edx/edx-platform/pull/11331
+        https://github.com/edx/edx-platform/pull/11611
+        """
+        self._create_library(org='admin_org_1', library='lib_adm_1', display_name='admin_lib_1')
+        self._create_library(org='admin_org_2', library='lib_adm_2', display_name='admin_lib_2')
+
+        self._login_as_non_staff_user()
+
+        self._create_library(org='staff_org_1', library='lib_staff_1', display_name='staff_lib_1')
+        self._create_library(org='staff_org_2', library='lib_staff_2', display_name='staff_lib_2')
+
+        with modulestore().default_store(ModuleStoreEnum.Type.split):
+            course = CourseFactory.create()
+
+        instructor_role = CourseInstructorRole(course.id)
+        auth.add_users(self.user, instructor_role, self.non_staff_user)
+
+        lib_block = ItemFactory.create(
+            category='library_content',
+            parent_location=course.location,
+            user_id=self.non_staff_user.id,
+            publish_item=False
+        )
+
+        def _get_libraries_to_select():
+            preview_url = '/xblock/{}/{}'.format(lib_block.location, STUDIO_VIEW)
+
+            resp = self.client.get_json(preview_url)
+            self.assertEquals(resp.status_code, 200)
+
+            parsed_response = parse_json(resp)
+            parser = etree.HTMLParser()
+            tree = etree.parse(StringIO(parsed_response['html']), parser)
+            divs = tree.xpath('/html/body/div/div[@id="settings-tab"]')
```

Comment:
This code works well, but is somewhat fragile, because it depends on the DOM layout (`'/html/body/div/div[@id="settings-tab"]'`) which is not necessarily stable. If someone re-organizes the XBlock settings editor UI HTML, it could break this test.

I think you should consider a much cruder approach, which is just as effective but a lot more resilient: Just get `editor_html = parse_json(resp)['html']`, then do a simple string search to assert that the name of each library is or is not in `editor_html`. Because the names `staff_lib_1` etc. are quite unique, there's basically no chance they'll be in the HTML unless they are in the library selector widget. So you don't have to parse the HTML in great detail.